bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -1050,25 +1050,11 @@ public void run() {
                                 .registerSuccessfulValue(batchSize);
 
                             boolean shouldRolloverJournal = (lastFlushPosition > maxJournalSize);
-                            if (syncData) {
```

Comment:
> is it expected ?

yes. that's expected. The flag `journalSyncData` was meant for any write do you need to wait for syncing journal data before triggering the callback.

> I think we should cover this change with some test, we already have some kind of advanced tests on journal flush, I added them while working on DEFERRED_SYNC.

I don't think we need any new tests. because existing tests already covered both cases. What we need to adjust here is to make sure those tests are aligned with the behavior fixed in this pull request.